### DISCUSSION TITLE
Need help with allowing access to code-server outside of localhost

### DISCUSSION THREAD

**Discussion by wisplite** [coder/code-server#4524] [Q&A] [answered]
I know this sounds ridiculous, it literally says not to do this in the docs, but hear me out:

My network is configured in such a way that there is a separate device handling reverse-proxy connections via caddy, and I need to be able to connect to code-server through the reverse proxy via that device. The only problem is I can't connect to it because code-server is restricted to just localhost. I can't configure caddy to run on the same server as code-server, because that would cause all my services to go down if the server crashes (which happens quite often).

Is there a way I could do a local reverse proxy (can't forward ports 80 or 443, as they're already being used, so I can't just have another caddy instance), or allow code-server to listen to local IPs as well as localhost?

**Comment by lg-nineoneninethreenine** [comment]
Hmmm... Try using Cloudflare or ngrok(use HTTP transfer if on ngrok, also the URL changes on every ngrok start so you need to note the URL)

**Reply by jsjoeio** [reply]
Big fan of `ngrok`. [`localtunnel`](https://www.npmjs.com/package/localtunnel) also works well

**Reply by nooobcoder** [reply]
@jsjoeio Wow, this localtunnel is too good, never heard before, definitely going to try that out too.

**Reply by spdk3125** [reply]
For me localtunnel was not working 
could you please help me with it

**Reply by rodhfr** [reply]
duckdns or cloudflare free dns + docker container to dinamic set ip is a solution

**Comment by wisplite** [comment]
From what I can tell ngrok, localtunnel, etc. can only forward externally, when I need the server to listen on the local IP so my existing reverse proxy on a separate computer can connect to it.

If they have the capability to do so, let me know, but I can’t tell for sure.

**Reply by jsjoeio** [reply]
Ooo...I see. This is beyond my knowledge scope. @code-asher is out next week but will be back and can hopefully help!

**Comment by code-asher** [comment, marked as answer]
You can make code-server listen on something other than localhost with `--bind-addr`.  For example `--bind-addr 0.0.0.0:8080` will bind to all interfaces.
